### PR TITLE
Item label enable when support

### LINKED ISSUE #389: Item labels are not grayed out when the editor field is disabled.
When a trait editor field is disabled (e.g. using `enabled_when`) the associated label widget should be also `disabled` so that it is obvious to the user that the field is not editable.

*Please note that this feature request is related to #298 because of the way labels are created in traitsui*

### PR DESCRIPTION
**Pull request by corranwebster**
Fixes #389.  With tests.

### CODE REVIEW

File: traitsui/tests/test_labels.py [hunk 1]
```diff
@@ -200,6 +220,48 @@ def test_qt_no_labels_on_the_right_bug():
...
+
+    with store_exceptions_on_all_threads():
+        dialog = EnableWhenDialog()
+        ui = dialog.edit_traits()
+
+        labelled_editor  = ui.get_editors('labelled_item')[0]
+        unlabelled_editor  = ui.get_editors('unlabelled_item')[0]
+        nose.tools.assert_is_none(unlabelled_editor.label_control)
+
+        nose.tools.assert_true(labelled_editor.label_control.isEnabled())
```
Comment (itziakos) [inline traitsui/tests/test_labels.py]: I prefer to have these two test functions into a single `unittest.TestCase`

Comment (corranwebster) [reply traitsui/tests/test_labels.py]: Some comment as previous PR - the current test uses nose and I don't want to mix nose and unittest in suites.

Comment (rkern) [reply traitsui/tests/test_labels.py]: You can also put them into a plain-old-class named `TestWhatever`. But I wouldn't worry about "mixing" `unittest` and `nose`. `TestCases` are perfectly good `nose` tests.

**Comment by corranwebster** [comment]
@itziakos Have simplified tests down to just one test function with common code.  Will worry about unittest conversion of test suite in separate PR later.

**Pull request merged**